### PR TITLE
Prototype Public Xml generation from JSON

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: app/models/purl.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
-class Purl < ApplicationRecord # rubocop:disable Metrics/ClassLength
+class Purl < ApplicationRecord
   has_and_belongs_to_many :collections
   has_many :constituent_memberships, inverse_of: :parent, dependent: :destroy
   has_many :constituents, -> { order 'constituent_memberships.sort_order' },
            through: :constituent_memberships, source: :child
+  has_many :parent_memberships, class_name: 'ConstituentMembership', inverse_of: :child, dependent: :destroy
```
Comment (justinlittman) [inline app/models/purl.rb:6]: Is this unrelated?

File: app/models/purl.rb [hunk 2]
```diff
@@ -90,7 +92,7 @@ def deleted?
   ##
   # Release tags where the value is true or is one of the default targets. If the object has been deleted, it retuns blank.
   # This is consumed by
-  # https://github.com/sul-dlss/searchworks_traject_indexer/blob/64359399e8f670ed414b1c56c648dc9b95ad6bad/lib/traject/readers/kafka_purl_fetcher_reader.rb#L26 # rubocop:disable Layout:LineLength
+  # https://github.com/sul-dlss/searchworks_traject_indexer/blob/64359399e8f670ed414b1c56c648dc9b95ad6bad/lib/traject/readers/kafka_purl_fetcher_reader.rb#L26:LineLength
```
Comment (justinlittman) [inline app/models/purl.rb:95]: Something seems to have gone wrong here.

File: app/services/cocina_object_store.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CocinaObjectStore
+  def self.find(druid)
+    path = "#{purl_druid_path(druid)}/cocina.json"
+    data_hash = JSON.parse(File.read(path))
+    Cocina::Models.build(data_hash)
```
Comment (justinlittman) [inline app/services/cocina_object_store.rb:5]: So let's suppose we're rolling out a breaking cocina change and re-publishing all items. What happens if we publish an object which requires loading objects that have not yet been republished? That is, are we requiring a republishing order?

File: app/models/purl.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/purl.rb:6]: No, this is how we get the constituents for virtual objects

Comment (justinlittman) [reply app/models/purl.rb:6]: Gotcha.

File: app/services/cocina_object_store.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/cocina_object_store.rb:5]: Only in the case of Virtual objects, where the constituents need to be published before the object that is composed of the constituents.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**